### DISCUSSION TITLE
RTL Support

### DISCUSSION THREAD

**Discussion by babakfp** [directus/directus#24277] [Feature Requests]
### Summary

The document direction should change accordingly when changing the Directus language to an RTL language. Currently, it stays LTR.

### Motivation

Without this feature, Directus is just unusable for RTL clients.

Imagine giving this to your clients (using RTL direction for an LTR language):

![image](https://github.com/user-attachments/assets/97bdff4d-0070-4ad3-af79-349a15282241)

Take a look at this:

![image](https://github.com/user-attachments/assets/bd53f33a-6116-46f4-b217-fd0e6063e416)

It looks like this: "ault language Def" instead of "Default language".

Related links:

Unfortunately, [this discussion](https://github.com/directus/directus/discussions/18355) is closed, and it's not possible to upvote it. Even if this new discussion gets the opportunity to get upvoted, not everyone from the previous discussion is going to know about and upvote it.

### Detailed Design

Let's say we have a CSS property like this `margin-right`. This should turn into `margin-left`.

You can use a build tool to take CSS styles and flip direction related properties/values. For example: https://www.npmjs.com/package/rtlcss.

Based on the language, Directus can load LTR or RTL styles. If the language is an RTL language, then load RTL styles. If the language is an LTR language, then load LTR styles. When unknown, default to original styles (LTR in this case).

IMPORTANT: there are fields/UIs that must always stay LTR, no matter if the language/direction is RTL. For example, a URL field like `https://docs.directus.io`. Other examples: code fields.

Using [`dir="auto"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir) can help with this.

![image](https://github.com/user-attachments/assets/42a762c0-05ab-4812-be0c-6fa7d288f687)

https://github.com/user-attachments/assets/3db33575-b916-4bee-8e67-daf90c4b3658

> `auto`, which lets the user agent decide. It uses a basic algorithm as it parses the characters inside the element until it finds a character with a strong directionality, then applies that directionality to the whole element.

The default is `"auto"` but in order to change the whole direction of the page, you need to use `dir="rtl"`, and then use `dir="auto"` to make it `"auto"` for some fields/places.

### Must Have:
- A way for the user to view the interface in RTL direction.

### Should Have:
- Auto detect the direction based on the language.

### Could Have:
- Maybe a switch field (On/Off) could be used so that the user can specify the direction if Directus can't figure it out. I'm not sure if such a scenario could happen.

### Drawbacks

I can't think of a negative impact. It can possibly be implemented in a single day.

### Alternatives

The developer can load custom CSS. This wouldn't be easy because things like `margin-right` would need to be handled manually.

NOTE: I'm new to Directus, so I'm not sure if it's possible to add custom CSS styles to Directus or not. It was said on the internet.

### Adoption Strategy

- "Is this a breaking change?": NO.
- "Can we write a migration?": No need.
- "How will this affect other projects in the Directus ecosystem?": It won't.

### Expected Result

![image](https://github.com/user-attachments/assets/5b3746a3-0124-42aa-8719-ae09ad5fabd2)

![image](https://github.com/user-attachments/assets/d74a1ff3-75b2-4bed-9676-8a0af1f1af17)

![image](https://github.com/user-attachments/assets/8736dbc8-43be-407b-8669-9777bc04bd1a)

![image](https://github.com/user-attachments/assets/ee4a9021-1b95-4414-a6fe-fed4b6ca96ed)

NOTE: I added custom font for Persian (Farsi) parts to make it look pretty.

**Comment by rijkvanzanten** [comment]
This is being worked on 🙂 #25494
